homeassistant/components/miele/light.py

```diff
@@ -0,0 +1,138 @@
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import AMBIENT_LIGHT, DOMAIN, LIGHT, LIGHT_OFF, LIGHT_ON, MieleAppliance
+from .coordinator import MieleConfigEntry, MieleDataUpdateCoordinator
+from .entity import MieleDevice, MieleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MieleLightDescription(LightEntityDescription):
+    """Class describing Miele light entities."""
+
+    value_fn: Callable[[MieleDevice], StateType]
+
+
+@dataclass
+class MieleLightDefinition:
+    """Class for defining light entities."""
+
+    types: tuple[MieleAppliance, ...]
+    description: MieleLightDescription
+
+
+LIGHT_TYPES: Final[tuple[MieleLightDefinition, ...]] = (
+    MieleLightDefinition(
+        types=(
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.COFFEE_SYSTEM,
+            MieleAppliance.HOOD,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.WINE_CABINET,
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+        ),
+        description=MieleLightDescription(
+            key="light",
+            value_fn=lambda value: value.state_light,
+            translation_key="light",
+        ),
+    ),
+    MieleLightDefinition(
+        types=(MieleAppliance.HOOD,),
+        description=MieleLightDescription(
+            key="ambient_light",
+            value_fn=lambda value: value.state_ambient_light,
+            translation_key="ambient_light",
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the light platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleLight(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in LIGHT_TYPES
+        if device.device_type in definition.types
+    )
+
+
+class MieleLight(MieleEntity, LightEntity):
+    """Representation of a Light."""
+
+    entity_description: MieleLightDescription
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id: str,
+        description: MieleLightDescription,
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator, device_id, description)
+        self.api = coordinator.api
```

Comment:
you set this in multiple platforms, would it make sense to move this to the entity.py and remove this constructor?

Reply:
Yes, I think it makes sense. However, I would like to do it in a follow up PR as it involves changes in all platforms but sensors and binary_sensors.